workflows/mapping/Snakefile

```diff
@@ -0,0 +1,266 @@
+# -*- snakemake -*-
+
+import yaml
+import itertools
+import pandas
+
+try:
+    config.update(yaml.load(open('config.yaml')))
+except IOError:
+    raise IOError('No "config.yaml" found; please supply another config '
+                  ' file using the --configfile argument')
+
+
+# ----------------------------------------------------------------------------
+# Helper functions
+#
+
+
+# srcdir() is a snakemake built-in function. WRAPPERS must be relative to this
+# file.
+HERE = srcdir('')
+WRAPPERS = '../../wrappers'
+def wrapper_for(tool):
+    return os.path.join(HERE, WRAPPERS, tool)
+
+
+def params_for(rule, key):
+    d = config.get('rules', {}).get(rule, {}).get('params', {}).get(key, '')
+    return d
+
+
+def threads_for(rule):
+    return config.get('rules', {}).get(rule, {}).get('threads', {})
+
+
+def fastq_screen_config_inputs():
+    """
+    Returns indexes and a table (as string) that can be written directly to
+    a config file.
+
+    Use indexes as the input to a rule; use the table to build a config file.
+    """
+    indexes = config['rules']['fastq_screen']['indexes']
+    aligner = config['rules']['fastq_screen']['aligner']
+    inputs = []
+    for label, prefix in indexes.items():
+        prefix = os.path.join(config['data_dir'], prefix)
+        inputs.extend(expand(prefix + '.{n}.bt2', n=[1, 2, 3, 4]))
+    return inputs
+
+
+sampletable = pandas.read_table(config['sampletable'], index_col=0)
+samples = sampletable.index.tolist()
+
```

Comment:
Can you come up with a set of required columns and how they should be named, based on the kinds of sampletables you currently have?